clang/test/SemaCXX/warn-unused-result.cpp

```diff
@@ -364,3 +364,21 @@ void id_print_name() {
     ((int(*)())f)();
 }
 } // namespace GH117975
+
+namespace BuildStringOnClangScope {
+
+[[clang::warn_unused_result("Discarded result")]]
+bool makeClangTrue() { return true; }
+
+[[gnu::warn_unused_result("Discarded result")]]
```

Comment:
I think it would be a kindness to give a warning here along the lines of "the GNU spelling of this attribute does not accept an argument" so users know why their message isn't being displayed at all (I don't think it should be a hard error as GCC does because that seems a bit user hostile but I could see a case for doing what they do with their spelling of the attribute). However, I don't think that's strictly required for this patch. WDYT @erichkeane ?